tests/test_magic_link/test_mock_magic_link.py

```diff
@@ -0,0 +1,63 @@
+from unittest import mock
+
+import pytest
+from app.notification.model.notifier import Notifier
+from app.notification.model.routes import send_email
+from tests.test_magic_link.magic_link_data import email_recipient_data
+from tests.test_magic_link.magic_link_data import expected_magic_link_data
+from tests.test_magic_link.magic_link_data import make_response_data
+
+
+# PLEASE REVIEW THIS TEST, IF IT IS OK THEN I CAN RE-WRITE ALL THE TESTS WITH MOCK DATA/RESPONSE. # noqa
+# TEST DOES NOT REQUIRE NETWORK.
+
+
+@pytest.mark.usefixtures("live_server")
+@mock.patch("app.notification.model.routes.request.get_json")
+@mock.patch("app.notification.model.routes.email_recipient")
+@mock.patch(
+    "app.notification.model.notifier.notifications_client.send_email_notification"  # noqa
+)
+@mock.patch("app.notification.model.routes.make_response")
+def test_mock_magic_link(
```

Comment:
🤔 💯 This is really cool Ram, Could you use whitespace to logically separate the different patches in this function? Also, I challenge you to see if there's a way to do this with fixtures - if there is then that will save you time in rewriting some tests :)

Reply:
Yeah, sure. I am already working on it.